### PR TITLE
Make reference_pixel index from zero

### PR DESCRIPTION
**Pull request by dstansby**
Part of the ongoing API consolidation.

### CODE REVIEW

**Comment by Cadair** [comment]
the tests are angry :D

File: sunpy/map/tests/test_mapbase.py [hunk 1]
```diff
@@ -635,9 +658,9 @@ def test_as_mpl_axes_aia171(aia171_test_map):
 
 def test_pixel_to_world_no_projection(generic_map):
     with pytest.warns(SunpyUserWarning, match='Missing metadata for observer'):
-        out = generic_map.pixel_to_world(*u.Quantity(generic_map.reference_pixel)+1*u.pix, origin=1)
-    assert_quantity_allclose(out.Tx, -10*u.arcsec)
-    assert_quantity_allclose(out.Ty, 10*u.arcsec)
+        out = generic_map.pixel_to_world(*u.Quantity(generic_map.reference_pixel))
+    assert_quantity_allclose(out.Tx, 0*u.arcsec)
+    assert_quantity_allclose(out.Ty, 0*u.arcsec)
```
Comment (dstansby) [inline sunpy/map/tests/test_mapbase.py:663]: Now the coordinate of the reference pixel is the same as the coordinate in the WCS header 🎉

Comment (Cadair) [reply sunpy/map/tests/test_mapbase.py:663]: well the issue was we were doing +1 and `origin=1` surely just `origin=1` would have been sufficient :facepalm:

File: sunpy/map/mapbase.py [hunk 2]
```diff
@@ -869,11 +870,17 @@ def reference_coordinate(self):
...
-                                       (self.meta.get('naxis1') + 1) / 2.) * u.pixel,
-                         self.meta.get('crpix2',
-                                       (self.meta.get('naxis2') + 1) / 2.) * u.pixel)
+        """
+        Pixel of reference coordinate.
+
+        The pixel returned uses zero-based indexing, so will be 1 pixel less
+        than the FITS CRPIX values.
+        """
+        #
```
Comment (Cadair) [inline sunpy/map/mapbase.py:879]: ?
```suggestion
```

File: sunpy/map/tests/test_mapbase.py [hunk 3]
```diff
@@ -427,8 +440,8 @@ def test_submap(generic_map, unit):
     submap = generic_map.submap([width / 2., height / 2.] * unit, [width, height] * unit)
 
     # Check to see if submap properties were updated properly
-    assert submap.reference_pixel.x.value == generic_map.meta['crpix1'] - width / 2.
-    assert submap.reference_pixel.y.value == generic_map.meta['crpix2'] - height / 2.
+    assert submap.reference_pixel.x.value == generic_map.meta['crpix1'] - 1 - width / 2.
+    assert submap.reference_pixel.y.value == generic_map.meta['crpix2'] - 1 -height / 2.
```
Comment (Cadair) [inline sunpy/map/tests/test_mapbase.py:444]:
```suggestion
    assert submap.reference_pixel.y.value == generic_map.meta['crpix2'] - 1 - height / 2.
```

**Pull request merged**